projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-details/query-details.component.ts

```diff
@@ -10,7 +11,16 @@ export class QueryDetailsComponent {
   @Input() public query: QueryListItem;
   @Output() public backClicked: EventEmitter<boolean> = new EventEmitter();
 
+  constructor(private sessionStorage: SessionStorageService) { }
+
   public onBack(): void {
     this.backClicked.emit(true);
   }
+
+  public isCaseworker(): boolean {
+    const userDetails = JSON.parse(this.sessionStorage.getItem('userDetails'));
+    return userDetails && userDetails.roles
```

Comment:
@DavidJayakumar - this can be simplified as below.
```
!(userDetails?.roles?.includes('pui-case-manager')
        || userDetails?.roles?.some((role) => role.toLowerCase().includes('judge')))
```